### DISCUSSION TITLE
Any chance of adding support for chatglm.cpp?

### DISCUSSION THREAD

**Discussion by mrdevolver** [LostRuins/koboldcpp#310] [Ideas]
Hi, any chance of adding support for [chatglm.cpp](https://github.com/li-plus/chatglm.cpp)?

You can find the GGML model [here](https://huggingface.co/Xorbits/chatglm2-6B-GGML) which is already a second version. It's been pretty popular on huggingface recently, it seems to be pretty fast and pretty good at chatting and story telling. Oh and it beats even some bigger models when it comes to using simple logic and math, something which would be very useful to make stories and games that actually make sense, you know?

**Comment by LostRuins** [comment]
Certainly a consideration to be added, but considering it's novel architecture, implementation will likely require significant time and effort, unless someone else is able to contribute it as a PR.

**Reply by mrdevolver** [reply]
Thank you for replying! At huggingface, I asked about possible support in KoboldCpp and I was told that KoboldCpp community will most likely not support chatglm in near future and I was recommended project called [Xinference](https://github.com/xorbitsai/inference) which is a simple web ui for using various models including chatglm models for chat. It's really barebones, not as suited for playing games as KoboldCpp and unfortunately for me, it only contains source code and I've been using KoboldCpp especially for availability of its precompiled Windows packages for a reason, but hey your skills are far ahead of my own and I thought maybe if you could take a look and see how it's implemented in Xinference, maybe that would make the work on implementation for KoboldCpp at least a bit easier. Well, just trying to help make the cool KoboldCpp even better.
